merlin/models/torch/sequences.py

```diff
@@ -0,0 +1,453 @@
+#
+# Copyright (c) 2023, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+from typing import Dict, List, Optional, Tuple, Union
+
+import torch
+import torch.nn.functional as F
+from torch import nn
+
+from merlin.models.torch.batch import Batch, Sequence
+from merlin.models.torch.schema import Selection, select
+from merlin.schema import ColumnSchema, Schema, Tags
+
+MASK_PREFIX = "__mask"
+
+
+class TabularPadding(nn.Module):
```

Comment:
I think it might be better to break padding out into a smaller PR first, and then do masking afterwards.